### DISCUSSION TITLE
CNCF TAG-Runtime Discussion/Presentation?

### DISCUSSION THREAD

**Discussion by raravena80** [wasmi-labs/wasmi#497] [General]
Hello wasmi team,

I'm one of the co-chairs of the CNCF SIG-Runtime, I'm reaching out and think it would be great for you to present/discuss the project at one of our meetings. An overview of the project would be great.

Let me know if this something you'd be interested in doing. If yes, please feel free to add it to our [agenda](https://bit.ly/cncf-sig-runtime-meeting-notes) or reach out to me (raravena80 at gmail.com)

Thanks!

**Comment by Robbepop** [comment]
@raravena80 Sorry for the late reply. What information do you guys need about `wasmi` and is this still needed?

**Comment by raravena80** [comment]
@Robbepop np. It would be great to have you present in our meeting.  A general overview of the project, its community and how it would fit cloud native applications. We would also be interested to hear how it's used in blockchain apps. How about Jan 20th or Feb 3rd? Thanks!

**Comment by Robbepop** [comment]
I am no expert in cloud computing myself so I'd need to know what the requirements of cloud computing are with respect to interpreters such as `wasmi` before I can provide more details about applicability.

**Comment by raravena80** [comment]
Can `wasmi` be used to test WASM modules? It's an interpreter right? i.e dry run (or prototype) before running the module in a production environment.    Additionally, are there any plans to support or provide a path to [WASI](https://wasi.dev/)?

**Comment by Robbepop** [comment]
Yes, `wasmi` is a Wasm interpreter and is able to execute Wasm modules. I am not entirely sure what exactly you mean with "dry running" but form what I understood the host side should be able to determine how the Wasm module executes and how state from the execution is synced or not.
We do not plan to implement WASI out of the box at the moment, however, it is very well possible for users of `wasmi` to provide their own WASI support using the existing `wasmi`.

In our own blockchain use case we only need `wasm32-unknown-unknown` and correct and deterministic executions are what matters to us.

**Comment by raravena80** [comment]
In any case it would be good to hear a general overview of the project and a demo.  Will that work?

**Comment by Robbepop** [comment]
At this point I cannot give a guarantee. Right now we are working on a big update for `wasmi` that will take some more time. It won't make sense to talk a lot about `wasmi` until that's done. I will get back to you once we are ready.

**Comment by raravena80** [comment]
Sounds good, thanks!

**Comment by Robbepop** [comment]
https://github.com/paritytech/wasmi/issues/358 is probably important for this. Can you confirm this @raravena80 ?

**Comment by raravena80** [comment]
Yes, correct.  Thanks!

**Comment by Robbepop** [comment]
@raravena80 I just converted this into a discussion item since from now on issues in the `wasmi` repository are reserved for open technicald discussions about `wasmi`. So this is a better fit for a GH Discussions forum post. Btw.: Is this still relevant to you? Apparently some external contributor volunteered to be working on the Wasi support for `wasmi`.

**Comment by raravena80** [comment]
@Robbepop yes, it's still relevant 😄 . It would be great to have someone from the community present.

**Reply by Robbepop** [reply]
> It would be great to have someone from the community present.

By this do you mean to give a presentation? If so could you please elaborate on what questions you might have and what especially is to your interest with respect to `wasmi`? What is your focus? What is your use case for it exactly? Why would you consider using `wasmi`? What are your expected work loads? etc.

**Comment by raravena80** [comment]
@Robbepop just a general overview of the project, its community, why it's useful, etc.  (Also, if interested how it would work together with the CNCF).  Let me know if you have any other questions.

**Reply by Robbepop** [reply]
> how it would work together with the CNCF

I am no cloud (computing) engineer and therefore have no much knowledge about cloud providers or CNCF. So to give you an elaborate answer as to how `wasmi` would work together with the CNCF and how useful it is to you I need answers to the questions I stated above.

- What are you planning on using `wasmi` for within the context of CNCF? E.g. what is the purpose of a Wasm exeuctor witihn CNCF?
- What are the expected workloads of a Wasm executor in the context of CNCF?
- What are your demands to a Wasm executor? E.g. with regards to determinism, memory usage, platform availability, required Wasm proposal support etc.)
- What are expected work loads in the context of `wasmi` usage in CNCF? How big are the expected Wasm modules? How long are Wasm modules executed at a time?; Long running Wasm processes or short running? etc.

**Comment by raravena80** [comment]
@Robbepop I just mentioned "how it would work with the CNCF" as an example of something that could be discussed (but not required 😄 )

To answer and provide more clarity:

- What are you planning on using wasmi for within the context of CNCF? E.g. what is the purpose of a Wasm exeuctor witihn CNCF?
  - Could it work with Kubernetes or together with any of the other CNCF projects? Is there any relationship? (Not a requirement but just a general question)
- What are the expected workloads of a Wasm executor in the context of CNCF?
  - Anything.   Edge workloads, cloud workloads, microservices, ML model related, etc, etc
- What are your demands to a Wasm executor? E.g. with regards to determinism, memory usage, platform availability, required Wasm proposal support etc.)
  - No demands really.  We just want to learn more about the project in general.
- What are expected work loads in the context of wasmi usage in CNCF? How big are the expected Wasm modules? How long are Wasm modules executed at a time?; Long running Wasm processes or short running? etc.
  - Nothing specific, it's pretty open. It could be short live workloads or long lived workloads. It could be large or small wasm modules. We just want to learn about the project 😄

**Comment by raravena80** [comment]
Hi @Robbepop checking in.  Would you be able to present/join one of our [upcoming meetings](bit.ly/cncf-tag-runtime-meeting-notes)?  Thx!

**Comment by Robbepop** [comment]
For demonstration purposes it would be nice if we first had a `wasmi` CLI app with basic WASI support. And from what I can say we are very close to having one!
